src/pool.rs

```diff
@@ -244,7 +244,31 @@ impl ConnectionPool {
                     let mut servers = Vec::new();
                     let mut replica_number = 0;
 
+                    // Load Mirror settings
                     for (address_index, server) in shard.servers.iter().enumerate() {
+                        let mut mirror_addresses: Vec<Address> = vec![];
+                        if let Some(mirror_settings_vec) = &shard.mirrors {
+                            for mirror_settings in mirror_settings_vec {
+                                if mirror_settings.mirroring_target_index != address_index {
+                                    continue;
+                                }
+                                mirror_addresses.push(Address {
+                                    id: address_id,
+                                    database: shard.database.clone(),
+                                    host: mirror_settings.host.clone(),
+                                    port: mirror_settings.port,
+                                    role: server.role,
+                                    address_index: 0,
```

Comment:
We send stats from the mirrors , so we should make sure these unique identifiers are unique.

Reply:
I think the stats use the `address_id`, not the `address_index`. The `address_id` is unique.

Reply:
Here is the `address_index` callsites
https://cs.github.com/levkk/pgcat?q=.address_index

They are both irrelevant to mirrors

Reply:
I followed the same pattern we do for the main address. For the server addresses, we set the `address_index` to equal  the index of the server in the configs array. We do the same for mirrors, we set the mirror address_index to be the index of the mirror in the mirror config array

https://github.com/levkk/pgcat/blob/main/src/pool.rs#L247-L254